frontend/context/StakingProgramProvider.tsx

```diff
@@ -84,6 +90,16 @@ export const StakingProgramProvider = ({ children }: PropsWithChildren) => {
 
   const { service } = useService(selectedService?.service_config_id);
 
+  const [defaultStakingProgramId, setDefaultStakingProgramId] = useState(
```

Comment:
should have a state here, because
1) when switch to another staking contract we navigate to the main page, should show relevant staking contract (even though the deployment is still in progress)
2) should update when switching between agent configs anyway, so added useEffect also

Reply:
updated the context comment